src/number_theory/zeta_values.lean

```diff
@@ -134,3 +147,214 @@ begin
+  ((-1) ^ (k + 1) * (2 * π) ^ (2 * k) / (2 * k)! * bernoulli_fun (2 * k) x),
+  { convert (has_sum_one_div_nat_pow_mul_fourier (by linarith : 2 ≤ 2 * k) hx),
+    { ext1 n,
+      rw [pow_mul (-1 : ℂ),neg_one_sq, one_pow, one_mul], },
+    { rw [pow_add, pow_one],
+      conv_rhs { rw [mul_pow], congr, congr, skip, rw [pow_mul, I_sq] },
+      ring, } },
+  convert ((has_sum_iff _ _).mp (this.div_const 2)).1,
+  { ext1 n,
+    convert (of_real_re _).symm,
+    rw of_real_mul,rw ←mul_div, congr,
+    { rw [of_real_div, of_real_one, of_real_pow], refl, },
+    { rw [of_real_cos, of_real_mul, fourier_coe_apply, fourier_coe_apply, cos, of_real_one, div_one,
+        div_one, of_real_mul, of_real_mul, of_real_bit0, of_real_one, int.cast_neg,
+        int.cast_coe_nat, of_real_nat_cast],
+      congr' 3,
+      { ring }, { ring }, }, },
+  { convert (of_real_re _).symm,
+    rw [of_real_mul, of_real_div, of_real_div, of_real_mul, of_real_pow, of_real_pow, of_real_neg,
+      of_real_nat_cast, of_real_mul, of_real_bit0, of_real_one],
+    ring },
+end
+
+lemma has_sum_one_div_nat_pow_mul_sin {k : ℕ} (hk : 1 ≤ k) {x : ℝ} (hx : x ∈ Ico (0:ℝ) 1) :
+  has_sum (λ n:ℕ, 1 / (n:ℝ) ^ (2 * k + 1) * real.sin (2 * π * n * x))
+  ((-1) ^ (k + 1) * (2 * π) ^ (2 * k + 1) / 2 / (2 * k + 1)! *
+  (polynomial.map (algebra_map ℚ ℝ) (polynomial.bernoulli (2 * k + 1))).eval x) :=
+begin
+  have : has_sum (λ n:ℕ, 1 / (n:ℂ) ^ (2 * k + 1) * (fourier n (x : 𝕌) - fourier (-n) (x : 𝕌)))
+  ((-1)^(k + 1) * I * (2 * π)^(2 * k + 1) / (2 * k + 1)! * bernoulli_fun (2 * k + 1) x),
+  { convert (has_sum_one_div_nat_pow_mul_fourier (by linarith : 2 ≤ 2 * k + 1) hx),
+    { ext1 n,
+      rw [pow_add (-1: ℂ), pow_mul (-1 : ℂ), neg_one_sq, one_pow, one_mul, pow_one,
+        ←neg_eq_neg_one_mul, ←sub_eq_add_neg], },
+    { rw [pow_add, pow_one],
+      conv_rhs { rw [mul_pow], congr, congr, skip, rw [pow_add, pow_one, pow_mul, I_sq] },
+      ring, }, },
+  convert ((has_sum_iff _ _).mp (this.div_const (2 * I))).1,
+  { ext1 n,
+    convert (of_real_re _).symm,
+    rw of_real_mul,rw ←mul_div, congr,
+    { rw [of_real_div, of_real_one, of_real_pow], refl, },
+    { rw [of_real_sin, of_real_mul, fourier_coe_apply, fourier_coe_apply, sin, of_real_one, div_one,
+        div_one, of_real_mul, of_real_mul, of_real_bit0, of_real_one, int.cast_neg,
+        int.cast_coe_nat, of_real_nat_cast, ←div_div, div_I, div_mul_eq_mul_div₀, ←neg_div,
+        ←neg_mul, neg_sub],
+      congr' 4,
+      { ring, }, { ring }, }, },
+  { convert (of_real_re _).symm,
+    rw [of_real_mul, of_real_div, of_real_div, of_real_mul, of_real_pow, of_real_pow, of_real_neg,
+      of_real_nat_cast, of_real_mul, of_real_bit0, of_real_one,
+      ←div_div, div_I, div_mul_eq_mul_div₀],
+    have : ∀ (α β γ δ : ℂ), α * I * β / γ * δ * I = I ^ 2 * α * β / γ * δ := by { intros, ring },
+    rw [this, I_sq],
+    ring, },
+end
+
+lemma zeta_value {k : ℕ} (hk : 1 ≤ k) : has_sum (λ n:ℕ, 1 / (n:ℝ) ^ (2 * k))
```

Comment:
I suggest that we prepend these names with `has_sum`. That way it's clear that this is not a formula of the form `zeta(k) = ...`. Of course it is morally the same.
```suggestion
lemma has_sum_zeta_value {k : ℕ} (hk : 1 ≤ k) : has_sum (λ n:ℕ, 1 / (n:ℝ) ^ (2 * k))
```

Reply:
I'd even go for `has_sum_zeta_value_of_even_nat`, to differentiate this from zeta value at negative integers, for instance.